app/src/Containers/SearchContainer.tsx

```diff
@@ -241,31 +240,35 @@ const SearchContainer = () => {
     return undefined
   }
 
+  
   const handleQuickSearchSubmit = (
     searchField: 'client' | 'description' | 'name' | 'role' | 'skills' | 'sector',
     searchQueryChoice: string,
   ) => {
     let searchQueries = [] as string[]
+    let results = [] as Projects
+
+    searchQueries.push(searchQueryChoice)
 
     if (searchField === 'role') {
       const relatedRoles = getRelatedRoles(searchQueryChoice)
 
       if (relatedRoles?.length) {
         searchQueries = relatedRoles
```

Comment:
Now that we've moved `searchQueries.push(searchQueryChoice)` up to line 251, `searchQueries` is no longer empty when we get to this line here, so we need to add to it rather than overwriting it.  So I think we should change this line here to `searchQueries = [...searchQueries, ...relatedRoles]` or `searchQueries = searchQueries.concat(relatedRoles)` (whichever you think is clearer)

Reply:
Thanks Joe! I've changed it to concat, using the parameter searchQuery rather than relatedRoles as it's not for searchField 'roles'. Hope that makes sense! It gets the desired results so that's good ;)

Reply:
Sorry Pam, I think I wasn't clear enough with my comment --

It was ok keeping line 251 as `searchQueries.push(searchQueryChoice)` -- this adds the quick search button choice (e.g. `'Web Developer'` or `'Scrum Master'`) to the `searchQueries` array.
(NB this should be `searchQueryChoice` which is the text on the quick search button, not `searchQuery` which is the text in the free search box.  Feel free to rename these variables if you want -- maybe they're not clear enough at the moment.  Also we don't want to use `.concat` here as that merges an array with another array, whereas we're adding a single item to the array hence using `.push`)

It's this line 254 which needs to be `searchQueries = [...searchQueries, ...relatedRoles]` or `searchQueries = searchQueries.concat(relatedRoles)` so that we add any related roles to the `searchQueries` array, not replace it entirely.

With the code as you have it right now, the `searchQueries` array is being overwritten on line 254, so it no longer includes the `searchQueryChoice` string e.g. `'Web Developer'` or `'Scrum Master'`.  We'll probably still mostly get the right results, but there are cases where we won't -- for example with `'Scrum Master'` where there are no related roles for that choice (so line 254 as you currently have it would make `searchQueries` an empty array in that case).  If we keep `searchQueryChoice` as part of the `searchQueries` array it should still work.  (Appreciate this might be hard to test at the moment as we don't have projects matching all the different roles.)

Hope that makes some sense(!) 🙂  Happy to chat through more if that's helpful.

Reply:
Ah, that makes way more sense! After replacing the line yesterday there must have been a cache error as it worked fine. Thanks a million for checking, have amended and tested on both android and ios.